client/src/App.tsx

```diff
@@ -4,27 +4,35 @@ import { Switch, Route } from 'react-router-dom';
 import EditProfile from './pages/EditProfile';
 import Header from './components/Header';
 import Home from './pages/Home';
-import ClassProfile from './pages/ClassProfile';
-import EditClasses from './pages/EditCourses';
+import CourseProfile from './pages/CourseProfile';
+import EditCourses from './pages/EditCourses';
 import Login from './pages/Login';
-import Register from './pages/Register';
 import AddCourse from './pages/AddCourse';
-import { LOGIN_ROUTE, REGISTER_ROUTE, EDIT_CLASSES_ROUTE, EDIT_PROFILE_ROUTE, TEMP_COURSE_ROUTE } from './Routes';
+import Register from './pages/Register';
+import { useDispatch } from 'react-redux';
+import { getCategories } from './redux/AppState/appSlice';
+import * as r from './Routes';
 
 const Container = styled.div``;
 
 function App() {
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(getCategories());
```

Comment:
how often does this get called? When they first visit the website? Wouldn't it make sense to only call this when we actually need to load the categories rather than keeping it in a global state?

Reply:
The categories are needed on the front page and most likely only need to get called once per visit. Is there a need to call it more than once?

Reply:
fair enough, guess not really unless by some slim chance someone doesn't refresh and we add more categories at that exact moment